### ISSUE TITLE
Thinkpad T450s show "Unable to find a synaptics device" when run `perl xSwipe.pl`

### ISSUE THREAD

**Issue by yimun** [iberianpig/xSwipe#37]
My system is Ubuntu 14.04, when I run 'perl xSwipe.pl', it shows like this：
```
➜  perl xSwipe.pl 
Smartmatch is experimental at xSwipe.pl line 121.
Unable to find a synaptics device.
Unable to find a synaptics device.

### init_synclient
Unable to find a synaptics device.
Unable to find a synaptics device.

### @area_setting: []
### $touchpad_size_h: 0
### $touchpad_size_w: 0
### $x_min_thredshould: 0
### $y_min_thredshould: 0
### $inner_edge_left: '0'
### $inner_edge_right: '0'
### $inner_edge_top: '0'
### $inner_edge_bottom: '0'

### $session_name: 'ubuntu'
Can't access shared memory area. SHMConfig disabled?
Unable to find a synaptics device.

### init
```
I doubt maybe something wrong with the touchpad driver. When input `xinput list` and it shows:
```
⎡ Virtual core pointer                    	id=2	[master pointer  (3)]
⎜   ↳ Virtual core XTEST pointer              	id=4	[slave  pointer  (2)]
⎜   ↳ PS/2 Synaptics TouchPad                 	id=9	[slave  pointer  (2)]
⎣ Virtual core keyboard                   	id=3	[master keyboard (2)]
    ↳ Virtual core XTEST keyboard             	id=5	[slave  keyboard (3)]
    ↳ Power Button                            	id=6	[slave  keyboard (3)]
    ↳ Video Bus                               	id=7	[slave  keyboard (3)]
    ↳ Sleep Button                            	id=8	[slave  keyboard (3)]
    ↳ Integrated Camera                       	id=10	[slave  keyboard (3)]
    ↳ AT Translated Set 2 keyboard            	id=11	[slave  keyboard (3)]
    ↳ ThinkPad Extra Buttons                  	id=13	[slave  keyboard (3)]
```

**Comment by philiparvidsson** [comment]
Same problem here. Compiled and installed driver (from Chosko). Touchpad stops working after login. Synclient says it's not loaded.

**Comment by iaxax** [comment]
Same problem!Hope to get a solution.Thanks.

**Comment by philiparvidsson** [comment]
Try the following upon login:

sudo modprobe -r psmouse
sudo modprobe psmouse

Works for me.

**Comment by iaxax** [comment]
Thanks much for your help!But it doesn't work for me.Here's the hint.

Smartmatch is experimental at ./xSwipe.pl line 121.
Unable to find a synaptics device.
Couldn't find synaptics properties. No synaptics driver loaded?

### init_synclient
Couldn't find synaptics properties. No synaptics driver loaded?
Couldn't find synaptics properties. No synaptics driver loaded?

### @area_setting: []
### $touchpad_size_h: 0
### $touchpad_size_w: 0
### $x_min_thredshould: 0
### $y_min_thredshould: 0
### $inner_edge_left: '0'
### $inner_edge_right: '0'
### $inner_edge_top: '0'
### $inner_edge_bottom: '0'

### $session_name: 'ubuntu'
Can't access shared memory area. SHMConfig disabled?
Couldn't find synaptics properties. No synaptics driver loaded?

### init

**Comment by philiparvidsson** [comment]
Have you installed the modified synaptics driver from Chosko?

**Comment by iaxax** [comment]
I just follow the README. Do you mean this?

 #### NOTE: If using Ubuntu14.04, or later
 ##### Install older version synaptics driver that is compatible with xSwipe.

$ sudo apt-get install -y git build-essential libevdev-dev autoconf automake libmtdev-dev xorg-dev xutils-dev libtool
$ sudo apt-get remove -y xserver-xorg-input-synaptics
$ git clone https://github.com/Chosko/xserver-xorg-input-synaptics.git
$ cd xserver-xorg-input-synaptics
$ ./autogen.sh
$ ./configure --exec_prefix=/usr
$ make
$ sudo make install

**Comment by philiparvidsson** [comment]
Yes. After that, using modprobe solves the issue for me as I described above. I cannot help you further than that. :-(

**Comment by iaxax** [comment]
OK.Anyway, I really appreciate for your help and patience!

**Comment by nonbinary** [comment]
For some machines, this will likely be linked to libinput. Several distros are moving away from synaptics drivers, and are starting to use libinput instead. I have the same issue on fedora 24, and I think it's because fedora no longer uses synaptics drivers, they use libinput instead. Not sure if xSwipe is compatible with libinput yet.

**Comment by yimun** [comment]
I upgrade my system to Ubuntu 14.04.1 and problem solved!
